src/mp/util.cpp

```diff
@@ -83,6 +85,9 @@ int SpawnProcess(int& pid, FdToArgsFn&& fd_to_args)
     }
 
     pid = fork();
+    if (pid == -1) {
+        throw std::system_error(errno, std::system_category(), "fork");
```

Comment:
Interesting, I didn't know system_error took a third string argument. It'd be good to start using this more places.

Reply:
Yes, it also occurred to me that if `std::system_category()` modifies `errno` then `std::system_error()` may get the wrong value of `errno`:
```
#include <errno.h>
#include <stdio.h>

int f1() {
    errno = 3;
    return 20;
}   

void f2(int e, int x) {
    printf("%d %d\n", e, x);
}   

int main(int, char**) {
    errno = 2; // resulted from fork() or whatever
    f2(errno, f1());
    return 0;
}
```
produces:
```
# clang 9.0.1
$ clang++90 t.cc -o t && ./t
2 20

# gcc 9.2.0
$ g++9 t.cc -o t && ./t
3 20
```

Reply:
It's a possible concern, but in practice if that constructor was setting errno, it would probably break a lot of things. Also in the worst case error message text would just be less clear

Reply:
Yes, safe to ignore in this case. Just the pattern `func1(x, func2_may_possibly_modify_x());` is better to be avoided.

Reply:
> system_error took a third string argument. It'd be good to start using this more places.

https://github.com/chaincodelabs/libmultiprocess/pull/23 Tell std::system_error() which function failed

Cheerz!